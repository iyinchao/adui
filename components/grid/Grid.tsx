import * as React from "react"
import Row from "./Row"
import Col from "./Col"

export interface IGrid {
  Row: typeof Row
  Col: typeof Col
}
/**
 * 栅格用于便捷搭建结构化的页面布局，常用于有明确功能分区的页面，如 Dashboard、项目卡片等。
 */
const Grid: IGrid = () => <div />

Grid.Row = Row
Grid.Col = Col

export default Grid
