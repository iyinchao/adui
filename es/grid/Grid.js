import * as React from "react";
import Row from "./Row";
import Col from "./Col";

var Grid = function Grid() {
  return React.createElement("div", null);
};

Grid.Row = Row;
Grid.Col = Col;
export default Grid;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIi4uLy4uL2NvbXBvbmVudHMvZ3JpZC9HcmlkLnRzeCJdLCJuYW1lcyI6WyJSZWFjdCIsIlJvdyIsIkNvbCIsIkdyaWQiXSwibWFwcGluZ3MiOiJBQUFBLE9BQU8sS0FBS0EsS0FBWixNQUF1QixPQUF2QjtBQUNBLE9BQU9DLEdBQVAsTUFBZ0IsT0FBaEI7QUFDQSxPQUFPQyxHQUFQLE1BQWdCLE9BQWhCOztBQVNBLElBQU1DLElBQVcsR0FBRyxTQUFkQSxJQUFjO0FBQUEsU0FBTSxnQ0FBTjtBQUFBLENBQXBCOztBQUVBQSxJQUFJLENBQUNGLEdBQUwsR0FBV0EsR0FBWDtBQUNBRSxJQUFJLENBQUNELEdBQUwsR0FBV0EsR0FBWDtBQUVBLGVBQWVDLElBQWYiLCJzb3VyY2VzQ29udGVudCI6WyJpbXBvcnQgKiBhcyBSZWFjdCBmcm9tIFwicmVhY3RcIlxuaW1wb3J0IFJvdyBmcm9tIFwiLi9Sb3dcIlxuaW1wb3J0IENvbCBmcm9tIFwiLi9Db2xcIlxuXG5leHBvcnQgaW50ZXJmYWNlIElHcmlkIHtcbiAgUm93OiB0eXBlb2YgUm93XG4gIENvbDogdHlwZW9mIENvbFxufVxuLyoqXG4gKiDmoIXmoLznlKjkuo7kvr/mjbfmkK3lu7rnu5PmnoTljJbnmoTpobXpnaLluIPlsYDvvIzluLjnlKjkuo7mnInmmI7noa7lip/og73liIbljLrnmoTpobXpnaLvvIzlpoIgRGFzaGJvYXJk44CB6aG555uu5Y2h54mH562J44CCXG4gKi9cbmNvbnN0IEdyaWQ6IElHcmlkID0gKCkgPT4gPGRpdiAvPlxuXG5HcmlkLlJvdyA9IFJvd1xuR3JpZC5Db2wgPSBDb2xcblxuZXhwb3J0IGRlZmF1bHQgR3JpZFxuIl19