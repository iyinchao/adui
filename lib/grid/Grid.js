"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var React = _interopRequireWildcard(require("react"));

var _Row = _interopRequireDefault(require("./Row"));

var _Col = _interopRequireDefault(require("./Col"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { "default": obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var Grid = function Grid() {
  return React.createElement("div", null);
};

Grid.Row = _Row["default"];
Grid.Col = _Col["default"];
var _default = Grid;
exports["default"] = _default;
//# sourceMappingURL=data:application/json;charset=utf-8;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIi4uLy4uL2NvbXBvbmVudHMvZ3JpZC9HcmlkLnRzeCJdLCJuYW1lcyI6WyJHcmlkIiwiUm93IiwiQ29sIl0sIm1hcHBpbmdzIjoiOzs7Ozs7Ozs7QUFBQTs7QUFDQTs7QUFDQTs7Ozs7Ozs7QUFTQSxJQUFNQSxJQUFXLEdBQUcsU0FBZEEsSUFBYztBQUFBLFNBQU0sZ0NBQU47QUFBQSxDQUFwQjs7QUFFQUEsSUFBSSxDQUFDQyxHQUFMLEdBQVdBLGVBQVg7QUFDQUQsSUFBSSxDQUFDRSxHQUFMLEdBQVdBLGVBQVg7ZUFFZUYsSSIsInNvdXJjZXNDb250ZW50IjpbImltcG9ydCAqIGFzIFJlYWN0IGZyb20gXCJyZWFjdFwiXG5pbXBvcnQgUm93IGZyb20gXCIuL1Jvd1wiXG5pbXBvcnQgQ29sIGZyb20gXCIuL0NvbFwiXG5cbmV4cG9ydCBpbnRlcmZhY2UgSUdyaWQge1xuICBSb3c6IHR5cGVvZiBSb3dcbiAgQ29sOiB0eXBlb2YgQ29sXG59XG4vKipcbiAqIOagheagvOeUqOS6juS+v+aNt+aQreW7uue7k+aehOWMlueahOmhtemdouW4g+WxgO+8jOW4uOeUqOS6juacieaYjuehruWKn+iDveWIhuWMuueahOmhtemdou+8jOWmgiBEYXNoYm9hcmTjgIHpobnnm67ljaHniYfnrYnjgIJcbiAqL1xuY29uc3QgR3JpZDogSUdyaWQgPSAoKSA9PiA8ZGl2IC8+XG5cbkdyaWQuUm93ID0gUm93XG5HcmlkLkNvbCA9IENvbFxuXG5leHBvcnQgZGVmYXVsdCBHcmlkXG4iXX0=