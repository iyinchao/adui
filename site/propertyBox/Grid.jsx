import React from "react"
import styles from "./components/propertyBox.scss"
import CodeBox from "./components/CodeBox"
import { Grid, Slider } from "../../components"

const numbers = { 0: 2, 1: 3, 2: 4, 3: 6 }
const gutters = { 0: 4, 1: 8, 2: 12, 3: 16, 4: 20, 5: 24 }

export default class PropertyBox extends React.Component {
  state = {
    number: 1,
    gutter: 3,
  }

  render() {
    const { number, gutter } = this.state

    const codes = `<Grid.Row
  gutter={${gutters[gutter]}}
>
  <Grid.Col span={${12 / numbers[number]}} />
  ...
</Grid.Row>`

    return (
      <div className={styles.wrapper}>
        <div className={styles.property}>
          <div className={styles.left} style={{ padding: "40px" }}>
            <Grid.Row style={{ flex: 1 }} gutter={gutters[gutter]}>
              {Array.from(new Array(numbers[number]), (_, i) => i).map(
                (item) => (
                  <Grid.Col span={12 / numbers[number]} key={item}>
                    <div
                      style={{
                        display: "flex",
                        alignItems: "center",
                        justifyContent: "center",
                        height: "160px",
                        fontFamily: "SFMono-Semibold",
                        fontSize: "13px",
                        lineHeight: "20px",
                        color: "#fff",
                        fontWeight: "600",
                        backgroundColor: "var(--primary-color)",
                      }}
                    >
                      Frame
                    </div>
                  </Grid.Col>
                )
              )}
            </Grid.Row>
          </div>
          <div className={styles.right}>
            <div className={styles.rightTitle}>可配置项</div>
            <dl className={styles.controls}>
              <dt className={styles.controlTitle}>栅格数量</dt>
              <dd className={styles.controlContent}>
                <Slider
                  inputVisible={false}
                  defaultValue={1}
                  onChange={(value) => {
                    this.setState({ number: value })
                  }}
                  min={0}
                  max={3}
                  marks={numbers}
                />
              </dd>
              <dt className={styles.controlTitle}>栅栏间距</dt>
              <dd className={styles.controlContent}>
                <Slider
                  inputVisible={false}
                  defaultValue={3}
                  onChange={(value) => {
                    this.setState({ gutter: value })
                  }}
                  min={0}
                  max={5}
                  marks={gutters}
                />
              </dd>
            </dl>
          </div>
        </div>
        <CodeBox codes={codes} />
      </div>
    )
  }
}
