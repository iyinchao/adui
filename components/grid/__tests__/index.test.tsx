import * as React from "react"
import TestRenderer from "react-test-renderer"
import Grid from ".."

describe("Grid", () => {
  it("是否正确渲染", () => {
    // @ts-ignore
    const grid = <Grid />
    const wrapper = TestRenderer.create(
      <div>
        {grid}
        <Grid.Row gutter={16}>
          <Grid.Col span={4} />
          <Grid.Col span={6} />
          <Grid.Col span={2} />
        </Grid.Row>
      </div>
    )
    expect(wrapper).toMatchSnapshot()
  })
})
