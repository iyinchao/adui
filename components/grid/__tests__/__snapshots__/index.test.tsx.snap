// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Grid 是否正确渲染 1`] = `
<div>
  <div />
  <div
    className="adui-grid-row"
    style={
      Object {
        "marginLeft": "-8px",
        "marginRight": "-8px",
      }
    }
  >
    <div
      className="adui-grid-col adui-grid-col_4"
      style={
        Object {
          "paddingLeft": "8px",
          "paddingRight": "8px",
        }
      }
    >
      
    </div>
    <div
      className="adui-grid-col adui-grid-col_6"
      style={
        Object {
          "paddingLeft": "8px",
          "paddingRight": "8px",
        }
      }
    >
      
    </div>
    <div
      className="adui-grid-col adui-grid-col_2"
      style={
        Object {
          "paddingLeft": "8px",
          "paddingRight": "8px",
        }
      }
    >
      
    </div>
  </div>
</div>
`;
